### PR TITLE
Выполненое 4 задание

### PR DESCRIPTION
**Pull request by Mishavint**
- [X] Найти данные размера хотя бы 10000, на которых можно решить задачу классификации или регрессии.
- [X] Выполнить разведочный анализ (**EDA**). При необходимости выполнить полезные преобразования данных.
- [X] Самостоятельно реализуйте алгоритмы **Bagging** и **AdaBoost**, которые позволят обучить ансамбли каких-то простых моделей, например деревьев принятия решений.
- [X] Обучите ансамбли для каких-то простых моделей, используя свои две реализации, а также реализации из **sklearn** (**Bagging**, **Random Forest**, **AdaBoost**, **Gradient Tree Boosting**, **Histogram-Based Gradient Boosting**).
- [X] При обучении использовать **подбор гиперпараметров**, **кросс-валидацию**.
- [X] Сравнить качество моделей: замерить скорость обучения, вычислить различные метрики качества, построить матрицы ошибок.
- [X] * (**+2 балла**) Также добавить в сравнение модели из библиотек **XGBoost**, **LightGBM** и **CatBoost**.
- [X] * (**+3 балла**) Для подбора гиперпараметров использовать фреймворк **Optuna**.

Выполнил Винцукевич Михаил. Группа 20.Б11-мм

### CODE REVIEW

**Review by rustam-azimov — APPROVED** [review]
10 баллов
